package/MDAnalysis/core/selection.py

```diff
@@ -172,30 +172,34 @@ def __init__(cls, name, bases, classdict):
 
 
 class LogicOperation(object, metaclass=_Operationmeta):
-    def __init__(self, lsel, rsel):
+    def __init__(self, lsel, rsel, parser):
         self.rsel = rsel
         self.lsel = lsel
+        self.parser = parser
+
+    def apply(self, *args, **kwargs):
+        return self._apply(*args, **kwargs).asunique(sorted=self.parser.sort)
```

Comment:
Sorry for the large diffs here -- having `.unique` in 21 different lines seemed repetitive, so I renamed them all to `_apply` and apply the uniqueness catch in `apply()`.